CryptoPrices/Sources/Data/Sources/NetworkCore/APIModels/APIPrices.swift

```diff
@@ -0,0 +1,13 @@
+//
+//  APIPrices.swift
+//  Data
+//
+//  Created by Minh Pham on 27/12/2022.
+//
+
+import Foundation
+
+public struct APIPrices: Decodable, Equatable {
+
+    public let prices: [[Double]]
+}
```

Comment:
For the price, I'd prefer to use the `Decimal` type than `Double`. Although our app does not involve any calculations so we don't have to worry about like miscalculation, rounding errors, ... but it's a good convention I think. And to make it consistent too since other types use `Decimal` already like `Coin` and `APICoin`.

To make the consumer of the network layer's life a little easier,  we can let the `CoinAPIProtocol` return a more usable types:
```swift
public protocol CoinAPIProtocol {

    func getChartPrices(coinID: String, numberOfDays: Int) async throws -> [APIPrice]
}
```
The remote API returns data in a format `prices: [[String]]`, so we can add an intermediate data type:
```swift
// Remote API returns
public struct APIPriceData: Decodable, Equatable {

    public let prices: [[Decimal]]
}
```

This is what the network layer returns:
```swift
// Our network layer returns (to Repository)
public struct APIPrice: Equatable {

    public let price: Decimal
    public let timestamp: Date

    public init(price: Decimal, timestamp: Date) {
        self.price = price
        self.timestamp = timestamp
    }
}
```

The implementation of the network layer just do some small extra work to convert to correct type:
```swift
extension Pilot: CoinAPIProtocol where R == CoinRoute {

    public func getChartPrices(coinID: String, numberOfDays: Int) async throws -> [APIPrice] {
        let priceData = try await request(
            .chart(GetChartPricesParameters(id: coinID, days: numberOfDays)),
            target: APIPriceData.self,  👈  Raw, original from remote sources
            decoder: .apiDecoder
        )

        return priceData.prices.map { prices in
            APIPrice(
                price: prices.first ?? 0.0,
                timestamp: prices.last.map { Date(timeIntervalSince1970: ($0 as NSDecimalNumber).doubleValue) } ?? Date()
            )
        }
    }
}
```
We can safely remove the `DataPoint` from the network layer 🎉

Reply:
One quick question, at the `CoinAPIProtocol` layer, shouldn't we return the exact response object and the mapping can be done via the `Repository` layer? I raised this because `APIPrice` is not an identical mapping to the response payload and thus it should not be treated as an `API` response IMO 🙏

On top of that, regarding the suggestion of the `APIPrice` class, I have the same concern as [this comment](https://github.com/nimblehq/nimble-crypto-ios/pull/75#discussion_r1058146716) 😅

> For the price, I'd prefer to use the Decimal type than Double. Although our app does not involve any calculations so we don't have to worry about miscalculations, or rounding errors, ... but it's a good convention I think. And to make it consistent too since other types use Decimal already like Coin and APICoin.

I have no issue with this and your reason makes sense to me, I will update 👍

Reply:
Update to use `Decimal` in [2378259](https://github.com/nimblehq/nimble-crypto-ios/pull/75/commits/2378259dd05125751a71af9148163c515fd36042) 🚀

Reply:
Yeah, totally agree with you here! The transformation should be made in the `Repository` layer 🙏